ビルドステップの一部変更

# github action でのエラー

   .................................................................................................................................................................................  
   ...........................................................................................................................done                                                    
   Setting IAM Policy...............................done                                                                                                                              
   Creating Revision...................................................................................................................................failed                         
   Deployment failed                                                                                                                                                                  
   ERROR: (gcloud.run.deploy) The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated          
   timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain   
   more information.                                                                                                                                                                  
                                                                                                                                                                                      
   Logs URL:                                                                                                                                                                          
   https://console.cloud.google.com/logs/viewer?project=se-auto-agent-20251222&resource=cloud_run_revision/service_name/sys-dev-automation/revision_name/sys-dev-automation-00003-8p  
   j&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22sys-dev-automation%22%0Aresource.labels.revision_name%3D%22sys-dev-automation-00003  
   -8pj%22                                                                                                                                                                            
   For more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start                                                                 
   Error: Process completed with exit code 1

# 原因とみているもの

ビルド時におけるメモリの使い果たし

# 対応
恒久対策: Dockerfileを修正し、TypeScriptを事前にJavaScriptへコンパイルするビルドステップを追加する。これによりメモリ使用量が大幅に削減されます。

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,18 +1,31 @@
+# Stage 1: Build
+FROM node:22-slim AS builder
+
+WORKDIR /app
+
+COPY package*.json ./
+RUN npm install
+
+COPY . .
+RUN npm run build
+
+# Stage 2: Run
 FROM node:22-slim
 
 WORKDIR /app
 
-# Install dependencies
+# Only copy production dependencies and built artifacts
 COPY package*.json ./
-RUN npm install
+RUN npm install --omit=dev
 
-# Copy source code
-COPY . .
+COPY --from=builder /app/dist ./dist
+# Copy other necessary runtime directories/files if any
+COPY prompts ./prompts
+COPY templates ./templates
+COPY docs ./docs
 
-# The app uses ts-node to run directly from source
 # Cloud Run will provide the PORT environment variable
 ENV PORT=8080
-
 EXPOSE 8080
 
 CMD ["npm", "start"]

--- a/package.json
+++ b/package.json
@@ -8,7 +8,9 @@
     "doc": "docs"
   },
   "scripts": {
-    "start": "NODE_OPTIONS='--loader ts-node/esm --no-warnings' npx ts-node src/index.ts",
+    "build": "tsc",
+    "start": "node dist/src/index.js",
+    "dev": "NODE_OPTIONS='--loader ts-node/esm --no-warnings' npx ts-node src/index.ts",
     "test": "npm run test:slack && npm run test:github && npm run test:feedback && npm run test:sp && npm run test:context",
     "test:slack": "NODE_OPTIONS='--loader ts-node/esm --no-warnings' npx ts-node tests/test_slack_integration.ts",
     "test:github": "NODE_OPTIONS='--loader ts-node/esm --no-warnings' npx ts-node tests/test_github_client.ts",

--- a/src/index.ts
+++ b/src/index.ts
@@ -114,7 +114,7 @@ app.event("app_mention", async ({ event, client, logger }: any) => {
   } catch (error) {
     logger.error(error);
     await client.chat.postMessage({
-      channel,
+      channel: event.channel,
       thread_ts: event.ts,
       text: `Error: Failed to process request. ${error instanceof Error ? error.message : String(error)}`,
     });
@@ -160,7 +160,7 @@ app.message(async ({ message, client, logger }: any) => {
   } catch (error) {
     logger.error(error);
     await client.chat.postMessage({
-      channel,
+      channel: message.channel,
       thread_ts: message.ts,
       text: `Error: Failed to process request. ${error instanceof Error ? error.message : String(error)}`,
     });

